uplink/decorators.py

```diff
@@ -103,10 +103,17 @@ class headers(MethodAnnotation):
     """
 
     def __init__(self, arg, **kwargs):
-        # TODO: allow the first argument to be a list or str, in
-        # which case you would split the strings by a colon delimiter
-        # to identify the headers.
-        self._headers = dict(arg, **kwargs)
+        if isinstance(arg, str):
+            key, value = map(str.strip, arg.split(":"))
+            self._headers = {key: value}
+        elif isinstance(arg, list):
+            headers = {}
+            for header in arg:
+                key, value = header.split(': ')
+                headers[key] = value
+            self._headers = headers
```

Comment:
Actually, you might as well move `map(str.strip, a.split(":")` into a separate method, since it's used here and also above. Then, you can simply do:
```python
self._headers = dict(map(self._get_header, arg))
```